src/group/group.manager.ts

```diff
@@ -0,0 +1,171 @@
+import mongoose from 'mongoose';
+import { IGroup } from './group.interface';
+import * as GroupRepository from './group.repository';
+import * as personManager from '../person/person.manager';
+import { ServerError } from '../utils/errors';
+import { StatusCodes } from 'http-status-codes';
+
+/**
+ * add a person to group if he doest exist in that group already
+ * @param {mongoose.Types.ObjectId} groupId - the id of the group where the person will be added
+ * @param {mongoose.Types.ObjectId} personId - the id of the person that will be added
+ * @returns {Promise<void>} - Resolves when the person is added successfully.
+ * @throws {GroupNotFoundError} - if there isnt a group with that id
+ * @throws {PersonAlreadyMemberError} - if the person already member in that group
+ */
+export const addPersonToGroupOnlyIfIsntExist = async (
+    groupId: mongoose.Types.ObjectId,
+    personId: mongoose.Types.ObjectId,
+) => {
+    const group = await GroupRepository.getgroups({ _id: groupId });
+
+    if (!group || group.length === 0) {
+        throw new ServerError(StatusCodes.NOT_FOUND, `Group not found with ID: ${groupId}`);
+    }
+
+    if (group[0].subGroups.includes(personId)) {
+        throw new ServerError(StatusCodes.BAD_REQUEST, `Person is already a member of the group with ID: ${groupId}`);
+    }
+
+    return GroupRepository.addPersonToGroup(groupId, personId);
+};
+
+/**
+ * add group to subgroup only if the group isnt already subgroup of that group and enforces all the rules.
+ * @param {mongoose.Types.ObjectId} groupId - the id of the group where the person will be added.
+ * @param {mongoose.Types.ObjectId} subGroupId - the id of the group that will be added as sub group.
+ * @returns {Promise<void>} - Resolves when the sub group is added successfully.
+ * @throws {GroupNotFoundError} - If the group with the specified ID is not found.
+ * @throws {GroupNotFoundError} - If the subgroup with the specified ID is not found.
+ * @throws {Error} - If the group and subgroup have the same ID (a group can't contain itself).
+ * @throws {GroupAlreadySubGroupError} - If the subgroup is already a subgroup of the group.
+ * @throws {Error} - If the group is an ancestor of the subgroup, creating a cyclic relationship.
+ */
+export const addSubGroupToGroupIfDoesntExist = async (
+    groupId: mongoose.Types.ObjectId,
+    subGroupId: mongoose.Types.ObjectId,
+) => {
+    const group = await GroupRepository.getgroups({ _id: groupId });
+    const subGroup = await GroupRepository.getgroups({ _id: subGroupId });
+
+    if (!group || group.length === 0) {
+        throw new ServerError(404, `Group not found with ID: ${groupId}`);
+    }
+    if (!subGroup || subGroup.length === 0) {
+        throw new ServerError(404, `Group not found with ID: ${groupId}`);
+    }
```

Comment:
Create errors based on the server error, like GroupNotFoundError(). the goal is to encapsulate the status and message.
```suggestion
    if (!group || group.length === 0) {
        throw new GroupNotFoundError();
    }
    if (!subGroup || subGroup.length === 0) {
        throw new GroupNotFoundError();
    }
```